pkg/image/api/validation/validation.go

```diff
@@ -59,9 +61,69 @@ func validateImage(image *api.Image, fldPath *field.Path) field.ErrorList {
 		}
 	}
 
+	for i, sig := range image.Signatures {
+		result = append(result, validateImageSignature(&sig, fldPath.Child("signatures").Index(i))...)
+	}
+
 	return result
 }
 
+func validateImageSignature(signature *api.ImageSignature, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(signature.Type) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("type"), ""))
+	}
+	if len(signature.Content) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("content"), ""))
+	}
+	if len(signature.ValidationResult) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("validationResult"), ""))
+	}
+
+	allowedValues := sets.NewString(
+		string(api.SignatureValidationResultUnknown),
+		string(api.SignatureValidationResultValid),
+		string(api.SignatureValidationResultRevoked),
+		string(api.SignatureValidationResultExpired),
+		string(api.SignatureValidationResultInvalid))
+	if !allowedValues.Has(string(signature.ValidationResult)) {
```

Comment:
These values are treated as an enumeration. We should be sure they won't have to be updated in the future.